sql/moz-fx-data-shared-prod/mozilla_vpn_derived/subscriptions_v1/query.sql

```diff
@@ -1,3 +1,147 @@
+CREATE TEMP FUNCTION udf_js_apple_receipt(receipt STRING)
+-- provide the full schema even though most of it is scrubbed
+RETURNS STRUCT<
+  environment STRING,
+  latest_receipt BYTES,
+  latest_receipt_info ARRAY<
+    STRUCT<
+      expires_date STRING,
+      expires_date_ms STRING,
+      expires_date_pst STRING,
+      in_app_ownership_type STRING,
+      is_in_intro_offer_period STRING,
+      is_trial_period STRING,
+      original_purchase_date STRING,
+      original_purchase_date_ms STRING,
+      original_purchase_date_pst STRING,
+      original_transaction_id STRING,
+      product_id STRING,
+      purchase_date STRING,
+      purchase_date_ms STRING,
+      purchase_date_pst STRING,
+      quantity STRING,
+      subscription_group_identifier STRING,
+      transaction_id STRING,
+      web_order_line_item_id STRING
+    >
+  >,
+  pending_renewal_info ARRAY<
+    STRUCT<
+      auto_renew_product_id STRING,
+      auto_renew_status STRING,
+      original_transaction_id STRING,
+      product_id STRING
+    >
+  >,
+  receipt STRUCT<
+    adam_id INT64,
+    app_item_id INT64,
+    application_version STRING,
+    bundle_id STRING,
+    download_id INT64,
+    in_app ARRAY<
+      STRUCT<
+        expires_date STRING,
+        expires_date_ms STRING,
+        expires_date_pst STRING,
+        in_app_ownership_type STRING,
+        is_in_intro_offer_period STRING,
+        is_trial_period STRING,
+        original_purchase_date STRING,
+        original_purchase_date_ms STRING,
+        original_purchase_date_pst STRING,
+        original_transaction_id STRING,
+        product_id STRING,
+        purchase_date STRING,
+        purchase_date_ms STRING,
+        purchase_date_pst STRING,
+        quantity STRING,
+        transaction_id STRING,
+        web_order_line_item_id STRING
+      >
+    >,
+    original_application_version STRING,
+    original_purchase_date STRING,
+    original_purchase_date_ms STRING,
+    original_purchase_date_pst STRING,
+    receipt_creation_date STRING,
+    receipt_creation_date_ms STRING,
+    receipt_creation_date_pst STRING,
+    receipt_type STRING,
+    request_date STRING,
+    request_date_ms STRING,
+    request_date_pst STRING,
+    version_external_identifier INT64,
+  >,
+  status INT64,
+> DETERMINISTIC
+LANGUAGE js
+AS
+  "return JSON.parse(receipt);";
```

Comment:
I'd like to see this is a udf with a validation document test, like we do mozilla-pipeline-schemas (as long as bigquery supports multtline strings).